src/openforms/forms/migrations/0094_convert_old_service_fetch_config.py

```diff
@@ -0,0 +1,86 @@
+# Generated by Django 4.2.10 on 2024-03-06 14:38
+from typing import TypedDict
+
+from django.db import migrations
+from django.db.models import QuerySet
+
+from ..models import FormLogic
+from ..constants import LogicActionTypes
+
+
+class UpdatedObjects(TypedDict):
+    rules: list
+    variables: list
+
+
+def convert_to_new_service_fetch_format(
+    rules: list[FormLogic], available_service_fetch_configs: QuerySet
+) -> UpdatedObjects:
+    updated_rules = []
+    updated_variables = []
+    for rule in rules:
+        rule_updated = False
+        for action in rule.actions:
+            if action["action"]["type"] != LogicActionTypes.fetch_from_service:
+                continue
+
+            if (service_fetch_config_pk := action["action"]["value"]) == "":
+                continue
+
+            action["action"]["value"] = ""
+            rule_updated = True
+            if not (
+                service_fetch_config := available_service_fetch_configs.filter(
+                    pk=service_fetch_config_pk
+                ).first()
+            ):
+                # Case in which the configuration is broken: the rule points to a non-existing service fetch config.
+                # We just empty the value and do nothing with the value it contained.
+                continue
+
+            variable = rule.form.formvariable_set.filter(key=action["variable"]).first()
+            if not variable:
+                # Another case of broken configuration.
+                # The variable does not exist, so we can't update its configuration for service fetch.
+                continue
+
+            variable.service_fetch_configuration = service_fetch_config
+            updated_variables.append(variable)
+            rule_updated = True
+
+        if rule_updated:
+            updated_rules.append(rule)
+
+    return UpdatedObjects(rules=updated_rules, variables=updated_variables)
```

Comment:
I think it's just personal preference, but a two-tuple `tuple[list[FormLogic], list[FormVariable]]` could avoid defining a typed dict.

The drawback is it's probably less readable. If we keep the typed dict however, the lists types can probably be parametrized with `list[FormLogic]` and `list[FormVariable]`